noir-projects/aztec-nr/aztec/src/utils/point.nr

```diff
@@ -1,17 +1,72 @@
 use dep::protocol_types::point::Point;
 
+// I am storing the modulus divided by 2 plus 1 here because full modulus would throw "String literal too large" error
+// Full modulus is 21888242871839275222246405745257275088548364400416034343698204186575808495617
+global BN254_FR_MODULUS_DIV_2: Field = 10944121435919637611123202872628637544274182200208017171849102093287904247808;
+
 /// Converts a public key to a byte array.
 ///
 /// We don't serialize the point at infinity flag because this function is used in situations where we do not want
 /// to waste the extra byte (encrypted log).
-pub fn pub_key_to_bytes(pk: Point) -> [u8; 64] {
-    assert(!pk.is_infinite, "Point at infinity is not a valid public key.");
-    let mut result = [0 as u8; 64];
-    let x_bytes = pk.x.to_be_bytes(32);
-    let y_bytes = pk.y.to_be_bytes(32);
-    for i in 0..32 {
-        result[i] = x_bytes[i];
-        result[i + 32] = y_bytes[i];
+pub fn point_to_bytes(pk: Point) -> [u8; 32] {
```

Comment:
This function nows compresses to 32 bytes instead of 64 just like we do in Point.toCompressedBuffer.

+ added a test